### ISSUE TITLE
GetCultureToUse-Method unclear

### ISSUE THREAD

**Issue by Compufreak345** [AlexTeixeira/Askmethat-Aspnet-JsonLocalizer#54]
I am currently trying to understand the ["GetCultureToUse"-method](https://github.com/AlexTeixeira/Askmethat-Aspnet-JsonLocalizer/blob/master/Askmethat.Aspnet.JsonLocalizer/Localizer/JsonStringLocalizerBase.cs#L39). I understand that it gets the localization from the MemCache which seems to be fine - what I don't understand are the calls to "SetCurrentCultureToCache" - they don't seem to have any effect as at the end of the method `SetCurrentCultureToCache(cultureToUse);` overrides the other 2 calls. Am I right to assume that only the last call is relevant and the other 2 can be removed? I also think the `else`part can be removed as it was done at the beginning.

```
        protected void GetCultureToUse(CultureInfo cultureToUse)
        {
            if (!_memCache.TryGetValue(GetCacheKey(cultureToUse), out localization))
            {
                if (_memCache.TryGetValue(GetCacheKey(cultureToUse.Parent), out localization))
                {
                    SetCurrentCultureToCache(cultureToUse.Parent);
                }
                else
                {
                    _memCache.TryGetValue(GetCacheKey(cultureToUse), out localization);
                    SetCurrentCultureToCache(_localizationOptions.Value.DefaultCulture);
                }
            }
            SetCurrentCultureToCache(cultureToUse);
        }
```

I think this method should do the same:

```
protected void GetCultureToUse(CultureInfo cultureToUse)
        {
            if (!_memCache.TryGetValue(GetCacheKey(cultureToUse), out localization))
            {
                _memCache.TryGetValue(GetCacheKey(cultureToUse.Parent), out localization));
            }
            SetCurrentCultureToCache(cultureToUse);
        }
```

**Comment by AlexTeixeira** [comment]
Hi,

GetCulture Get the culture to use base on the CurrentUICulture.

In this comment : https://github.com/AlexTeixeira/Askmethat-Aspnet-JsonLocalizer/issues/52#issuecomment-492104806

We discussed the way to get the value of a localized key. The method has the role to get the dictionary from the cache base on these rules.

This allows plugin to refresh localized value without refreshing the thread for example.
You can have more information in this thread about the origin of this change 

https://github.com/AlexTeixeira/Askmethat-Aspnet-JsonLocalizer/issues/48

**Comment by Compufreak345** [comment]
Thanks for the explanation, so I think the third call to the memCache should access the DefaultCulture, not the CurrentCulture, and SetCurrentCultureToCache should be called with the used culture.

So my implementation is:

```
protected void GetCultureToUse(CultureInfo cultureToUse)
        {
            if (MemCache.TryGetValue(GetCacheKey(cultureToUse), out Localization))
            {
                SetCurrentCultureToCache(cultureToUse);
                return;
            }

            if (MemCache.TryGetValue(GetCacheKey(cultureToUse.Parent), out Localization))
            {
                SetCurrentCultureToCache(cultureToUse.Parent);
                return;
            }

            if (MemCache.TryGetValue(GetCacheKey(LocalizationOptions.Value.DefaultCulture), out Localization))
            {
                SetCurrentCultureToCache(LocalizationOptions.Value.DefaultCulture);
            }
        }
```

**Comment by AlexTeixeira** [comment]
If test are all green and if not regression in #48. It's ok for me :)

**Comment by Compufreak345** [comment]
All tests green now and the sample-project is still working fine - so I think #48 should not occur as well.

With your current implementation there might occur an cache mixup if I understand it correctly.

**Comment by AlexTeixeira** [comment]
Associated pullrequest #55

**Comment by AlexTeixeira** [comment]
@Compufreak345 changes available in 2.1.0-preview2

**Comment by AlexTeixeira** [comment]
Issues closed after no feedback
